### PR TITLE
Feat/reuse buffer serialization

### PR DESCRIPTION
**Pull request by J-Loudet**
The purpose of this PR is to minimize the number of allocations performed when
serializing data. For this end, the connector and built-in Source now create
internal `Vec<u8>` buffers that are reused whenever a message is serialized.

The name and signature of the impacted methods where changed accordingly.

This change should hopefully improve performance.

### CODE REVIEW

**Comment by gabrik** [comment]
Rebase on latest master please.

File: zenoh-flow/src/io/output.rs [hunk 1]
```diff
@@ -190,14 +190,14 @@ impl OutputBuilder {
     /// ```
     pub fn build_typed<T: Send + Sync + 'static>(
         self,
-        serializer: impl Fn(&T) -> anyhow::Result<Vec<u8>> + Send + Sync + 'static,
+        serializer: impl Fn(&mut Vec<u8>, &T) -> anyhow::Result<()> + Send + Sync + 'static,
```
Comment (gabrik) [inline zenoh-flow/src/io/output.rs:193]: Why not a slice `&mut [u8]` ?

If it is because it cannot grow then I'm fine with it.

**Review by gabrik — APPROVED** [review]
LGTM, just a minor comment, that I think I have answered myself.

**Comment by J-Loudet** [comment]
Wrong force-push order, closing & re-opening for clarity

**Pull request closed without merging**